packages/erk-shared/src/erk_shared/integrations/pr/submit.py

```diff
@@ -0,0 +1,199 @@
+    if has_issue_reference(impl_dir):
+        issue_ref = read_issue_reference(impl_dir)
+        if issue_ref is not None:
+            issue_number = issue_ref.issue_number
+            yield ProgressEvent(f"Found linked issue: #{issue_number}")
+
+    # Step 6: Push branch to remote
+    yield ProgressEvent("Pushing branch to origin...")
+    ops.git.push_to_remote(cwd, "origin", branch_name, set_upstream=True)
+    yield ProgressEvent("Branch pushed to origin", style="success")
+
+    # Step 7: Check for existing PR
+    yield ProgressEvent("Checking for existing PR...")
+    existing_pr = ops.github.get_pr_for_branch(repo_root, branch_name)
+
+    if isinstance(existing_pr, PRNotFound):
+        # Create new PR
+        yield ProgressEvent("Creating new PR...")
+
+        # Build PR body with footer
+        footer = build_pr_body_footer(
+            pr_number=0,  # Will be updated after creation
+            issue_number=issue_number,
+        )
+        full_body = pr_body + footer
+
+        pr_number = ops.github.create_pr(
+            repo_root,
+            branch=branch_name,
+            title=pr_title,
+            body=full_body,
+            base=trunk_branch,
+        )
+
+        # Get PR URL
+        pr_details = ops.github.get_pr(repo_root, pr_number)
+        if isinstance(pr_details, PRNotFound):
+            # This shouldn't happen but handle gracefully
+            pr_url = f"https://github.com/{branch_name}/pull/{pr_number}"
+        else:
+            pr_url = pr_details.url
+
+        # Update footer with actual PR number
+        updated_footer = build_pr_body_footer(
+            pr_number=pr_number,
+            issue_number=issue_number,
+        )
+        updated_body = pr_body + updated_footer
+        ops.github.update_pr_body(repo_root, pr_number, updated_body)
+
+        yield ProgressEvent(f"Created PR #{pr_number}", style="success")
+        yield CompletionEvent(
+            CoreSubmitResult(
+                success=True,
+                pr_number=pr_number,
+                pr_url=pr_url,
+                branch_name=branch_name,
+                issue_number=issue_number,
+                was_created=True,
+                message=f"Created PR #{pr_number}",
+            )
+        )
+    else:
+        # PR exists, just update if needed
+        pr_number = existing_pr.number
+        pr_url = existing_pr.url
+        yield ProgressEvent(f"Found existing PR #{pr_number}")
+
+        # Update PR body with footer (ensure checkout command and issue closing are present)
+        footer = build_pr_body_footer(
+            pr_number=pr_number,
+            issue_number=issue_number,
+        )
+        # Get current body and update if needed
+        current_body = ops.github.get_pr_body(repo_root, pr_number)
+        if current_body is None:
+            current_body = ""
+
+        # Check if footer already present (by looking for checkout command)
+        if "erk pr checkout" not in current_body:
```

Comment:
wrap this in a function so we can evolve "has_body_footer" logic over time

Reply:
Resolved - Extracted has_body_footer() function to allow logic evolution

_Addressed via `/erk:pr-address` at 2025-12-11 06:37 EST_